pilot/cmd/pilot-agent/main.go

```diff
@@ -127,6 +126,9 @@ var (
 	enableGatewaySDSEnv = env.RegisterBoolVar("ENABLE_INGRESS_GATEWAY_SDS", false,
 		"Enable provisioning gateway secrets. Requires Secret read permission").Get()
 
+	trustDomainEnv = env.RegisterStringVar("TRUST_DOMAIN", "cluster.local",
```

Comment:
This was removed in #27418 which is backwards incompatible, we have users relying on setting this. The flag should be removed instead, as done in this PR, IMO.

Reply:
we ignore them